bash-installer/installer/cassandra-diagnostics-jvm-opts-manager.sh

```diff
@@ -0,0 +1,170 @@
+
+    if [ $start_line -eq -1 ] && [ $end_line -gt 0 ]; then
+        exit_with_invalid_configuration
+    fi
+
+    if [ $end_line -eq -1 ] && [ $start_line -gt 0 ]; then
+        exit_with_invalid_configuration
+    fi
+
+    if [ $start_line -ge $end_line ]; then
+        exit_with_invalid_configuration
+    fi
+
+    [ -n "$start_line" ] && [ -n "$end_line" ]
+}
+
+function exit_with_invalid_configuration() {
+    print_error "Cassandra-diagnostics configuration in $1 seems to be invalid."
+    print_error "Please fix it manually and re-run installer."
+
+    exit $INVALID_CONFIGURATION_EXIT_CODE
+}
+
+# Finds line number where cassandra-diagnostics JVM options header is in input file.
+#
+# Parameters:
+#   $1 - file to search for header.
+#
+# Returns:
+#   - line number where header is, if found.
+#   - "-1" if header is not found.
+function find_header_line_number_in() {
+    echo $(find_line_number_in "$1" "$DIAGNOSTICS_JVM_OPTIONS_HEADER")
+}
+
+# Finds line number where cassandra-diagnostics JVM options footer is in input file.
+#
+# Parameters:
+#   $1 - file to search for footer.
+#
+# Returns:
+#   - line number where footer is, if found.
+#   - "-1" if footer is not found.
+function find_footer_line_number_in() {
+    echo $(find_line_number_in "$1" "$DIAGNOSTICS_JVM_OPTIONS_FOOTER")
+}
+
+# Finds line number of provided string in provided file.
+#
+# Parameters:
+#   $1 - file to search.
+#   $2 - string being searched.
+#
+# Returns:
+#   - line number where string is, if found.
+#   - "-1" if string is not found.
+function find_line_number_in() {
+    local line_number=$(grep -n "$2" "$1" | cut -f1 -d:)
+
+    if [ -z "$line_number" ]; then
+        echo -1
+    else
+        echo "$line_number"
+    fi
+}
+
+# Removes cassandra-diagnostics JVM options from provided file.
+#
+# Parameters:
+#   $1 - file from which to remove options.
+function remove_diagnostics_configuration_from() {
+    let local start_line=$(find_header_line_number_in "$1")
+    let local end_line=$(find_footer_line_number_in "$1")
+
+    delete_lines_from "$1" $start_line $end_line
+}
+
+# Deletes lines from file.
+#
+# Parameters:
+#   $1 - file from which to delete lines.
+#   $2 - start line number.
+#   $3 - end line number.
+function delete_lines_from() {
+    local cassandra_env_file_path="$1"
+    local start_line="$2"
+    local end_line="$3"
+
+    sed -i"$CASSANDRA_ENV_SCRIPT_BACKUP_EXTENSION" "$start_line,$end_line d" "$cassandra_env_file_path"
+}
+
+# Appends cassandra-diagnostics JVM options to provided file.
+#
+# Parameters:
+#   $1 - file to append JVM options to.
+#   $2 - absolute path to cassandra-diagnostics configuration file.
+function append_managed_configuration() {
+    local cassandra_env_file_path="$1"
+    local absolute_conf_file_path="$2"
+
+    echo -e "$DIAGNOSTICS_JVM_OPTIONS_HEADER \n" >> "$cassandra_env_file_path"
+    echo -e "$DIAGNOSTICS_JVM_OPTIONS_TEMPLATE""$absolute_conf_file_path \n" >> "$cassandra_env_file_path"
```

Comment:
JVM_OPTS="$JVM_OPTS -javaagent:$CASSANDRA_HOME/lib/cassandra-diagnostics-core-VERSION.jar" missing here. This triggers initialization of cassandra diagnostics